homeassistant/components/statistics/sensor.py

```diff
@@ -738,22 +731,22 @@ async def _initialize_from_database(self) -> None:
             self.async_write_ha_state()
         _LOGGER.debug("%s: initializing from database completed", self.entity_id)
 
-    def _update_attributes(self) -> None:
+    def _update_extra_state_attributes(self) -> None:
         """Calculate and update the various attributes."""
+        self._attr_extra_state_attributes[STAT_BUFFER_USAGE_RATIO] = None
+        self._attr_extra_state_attributes[STAT_AGE_COVERAGE_RATIO] = None
         if self._samples_max_buffer_size is not None:
-            self.attributes[STAT_BUFFER_USAGE_RATIO] = round(
+            self._attr_extra_state_attributes[STAT_BUFFER_USAGE_RATIO] = round(
                 len(self.states) / self._samples_max_buffer_size, 2
             )
 
         if self._samples_max_age is not None:
             if len(self.states) >= 1:
-                self.attributes[STAT_AGE_COVERAGE_RATIO] = round(
+                self._attr_extra_state_attributes[STAT_AGE_COVERAGE_RATIO] = round(
                     (self.ages[-1] - self.ages[0]).total_seconds()
                     / self._samples_max_age.total_seconds(),
                     2,
                 )
-            else:
-                self.attributes[STAT_AGE_COVERAGE_RATIO] = None
```

Comment:
Could be replace by `0`

Reply:
You mean setting it to `0` instead of `None` when it doesn't compute?
I don't know if that makes sense given we don't compute by age?

Reply:
Continuing your example: The automation would not make sense any more. Wouldn't the user want this automation to fail loudly, so they can adjust this function? There is nothing worse than a silently failing safety/sanity check.

Reply:
GitHub doesn't allow code change suggestions on deleted lines. (Strange)

```
                self.attributes[STAT_AGE_COVERAGE_RATIO] = 0
```

You will have to adopt 1-2 test cases.

Reply:
> Continuing your example: The automation would not make sense any more. Wouldn't the user want this automation to fail loudly, so they can adjust this function? There is nothing worse than a silently failing safety/sanity check.

I know the example wasn't great. Also as this isn't changing on runtime I can also see the reason this doesn't need to change. Maybe @emontnemery has a view here?

Reply:
Closing this as resolved, since it will now display (as before) during the runtime of the entity.